addons/point_of_sale/static/src/js/models.js

```diff
@@ -1356,6 +1356,13 @@ class Product extends PosModel {
         }
         return this.pos.units_by_id[unit_id];
     }
+    isPricelistItemUsable(item, context) {
+        return (
+            (!item.categ_id || _.contains(context[0], item.categ_id[0])) &&
+            (!item.date_start || moment.utc(item.date_start).isSameOrBefore(context[1])) &&
+            (!item.date_end || moment.utc(item.date_end).isSameOrAfter(context[1]))
+        );
+    }
```

Comment:
To be fair, this newly method doesn't need to have a `context` parameter since both `date` and `category_ids` are only used in it. You can directly put them here.

Reply:
Also, in JavaScript `obj[index]` doesn't work, you should do `obj[key]` (cf: `context[0]`, `context[1]`)